test/langtools/tools/javac/sym/ElementStructureTest.java

```diff
@@ -515,7 +515,11 @@ private void writeConstant(Object value) throws IOException {
             } else if (value instanceof Float) {
                 out.write(Float.toHexString((Float) value));
             } else {
-                out.write(String.valueOf(value));
+                if (value instanceof Character ch && Character.isSurrogate(ch)) {
```

Comment:
Looks like one would want to match the style, and of `else if`, joining the if-chain above?